File: ApiGateway/src/main/java/ch/emf/youquiz/beans/Reponse.java
package ch.emf.youquiz.beans;

public class Reponse {

    private Integer pkReponse;
    private String nom;
    private Boolean correct;

    public Reponse(Integer pkReponse, String nom, Boolean correct) {
        this.pkReponse = pkReponse;
        this.nom = nom;
        this.correct = correct;
    }

    // Getters and setters...
    public Integer getPkReponse() {
        return pkReponse;
    }

    public void setPkReponse(Integer pkReponse) {
        this.pkReponse = pkReponse;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public Boolean isCorrect() {
        return this.correct;
    }

    public void setCorrect(Boolean correct) {
        this.correct = correct;
    }
}

File: Client2/js/beans/Reponse.js
class Reponse {
    constructor(pkReponse, nom, correct) {
        this.pkReponse = pkReponse || null;
        this.nom = nom || '';
        this.correct = correct || false;
    }

    static fromJSON(reponseJSON) {
        return new Reponse(reponseJSON.pkReponse, reponseJSON.nom, reponseJSON.correct);
    }

    // Getters and setters...
    getPkReponse() {
        return this.pkReponse;
    }

    setPkReponse(pkReponse) {
        this.pkReponse = pkReponse;
    }

    getNom() {
        return this.nom;
    }

    setNom(nom) {
        this.nom = nom;
    }

    isCorrect() {
        return this.correct;
    }

    setCorrect(correct) {
        this.correct = correct;
    }
}
